lib/ecto/query.ex

```diff
@@ -1928,17 +1967,41 @@ defmodule Ecto.Query do
 
   If `limit` is given twice, it overrides the previous value.
 
+  ## Options
+
+    * `:with_ties` - a boolean controlling whether records tied for the last
+       spot will be returned. Not every adapter will support this option and
+       some adapters may require a corresponding `order_by` clause. Please 
+       check your database's documentation for more details. For SQL databases,
+       it may be found under the `FETCH` command instead of `LIMIT`. Defaults to `false`
+
   ## Keywords example
 
       from(u in User, where: u.id == ^current_user, limit: 1)
 
+      from(u in User, where: u.id == ^current_user, limit: 1, with_ties: true)
+
   ## Expressions example
 
       User |> where([u], u.id == ^current_user) |> limit(1)
 
+      User |> where([u], u.id == ^current_user) |> limit([], 1, with_ties: true)
+
   """
-  defmacro limit(query, binding \\ [], expr) do
-    Builder.LimitOffset.build(:limit, query, binding, expr, __CALLER__)
+  defmacro limit(query, binding \\ [], expr, opts \\ [])
```

Comment:
There is another approach here, which is to introduce a `with_ties` macro and it can only be called if a `limit` is set. It is similar to the `recursive_ctes` approach. :) I think it may be cleaner?

Reply:
That would be a lot cleaner. Was tough trying to fit this into the existing macros.

The downside I see is using it in kw queries, you couldn't do this: `limit: 5, with_ties: true`. That seemed like the most natural way to use it to me. But yeah it doesn't seem there's a perfect solution either way.

Reply:
Ahh, I get it. Sorry for being dense. Then basically keep limit as it was and add a `:with_ties` field to the query struct?

Reply:
I would still define LimitQuery. But you may be able to keep using the `LimitOffset` builder if you pass the struct you want it to build as argument.